include/curl/websockets.h

```diff
@@ -30,13 +30,20 @@ extern "C" {
 
 struct curl_ws_frame {
   int age;              /* zero */
+  int actual_flags;      /* See the CURLWS_FLAGS_* defines */
```

Comment:
Clearly this is not the final variable name. This is just to distinguish it from the existing `flags` variable.

What is currently in `flags` is actually the opcode and the offset flag. The opcode is not comprised of flags so I suggest we change the `CURLWS_TEXT`, `CURLWS_BINARY` and so on "flags" so be their actual opcode values. Having them as flags seems very confusing to me and makes callers think they can be ORed together which AFAICS they can't. In fact I think it is impossible to send a fragmented message but I'll go into that in a further email on the libcurl mailing list.

Further work then is to rename `flags` as `opcode` and deal with the `CURLWS_OFFSET` flag separately.